tests/Model/Message/MessageBagTest.php

```diff
@@ -103,18 +103,23 @@ public function withoutSystemMessage(): void
         $messageBag = new MessageBag(
             Message::forSystem('My amazing system prompt.'),
             Message::ofAssistant('It is time to sleep.'),
+            Message::forSystem('A system prompt in the middle.'),
             Message::ofUser('Hello, world!'),
+            Message::forSystem('Another system prompt at the end'),
         );
 
         $newMessageBag = $messageBag->withoutSystemMessage();
 
-        self::assertCount(3, $messageBag);
+        self::assertCount(5, $messageBag);
         self::assertCount(2, $newMessageBag);
 
-        $messageFromNewBag = $newMessageBag->getMessages()[0];
+        $assistantMessage = $newMessageBag->getMessages()[0];
+        self::assertInstanceOf(AssistantMessage::class, $assistantMessage);
+        self::assertSame('It is time to sleep.', $assistantMessage->content);
 
-        self::assertInstanceOf(AssistantMessage::class, $messageFromNewBag);
-        self::assertSame('It is time to sleep.', $messageFromNewBag->content);
+        $userMessage = $newMessageBag->getMessages()[1];
+        self::assertInstanceOf(UserMessage::class, $userMessage);
+        self::assertSame('Hello, world!', $userMessage->content[0]->text);
```

Comment:
Any clue how to get rid of this PHPStan error @chr-hertel ?

Reply:
I think it would be no fautl to add phpstan-phpunit extension. Then it should detect that you checked the instance of before accessing the content. Or you have to add an instanceof assertion for the index key of the content[0].

Reply:
in this case the assert was enough :+1: